test/tests/props.js

```diff
@@ -82,7 +82,56 @@ describe('zoid props cases', () => {
                     }
                 })
             });
-            
+
+            return instance.render(getBody());
+        });
+    });
+
+    it('should merge props definition with input props before passing to decorator', () => {
+        return wrapPromise(({ expect }) => {
+            window.__component__ = () => {
+                return zoid.create({
+                    tag:    'test-prop-value-container-url-param-bug',
+                    url:    'mock://www.child.com/base/test/windows/child/index.htm',
+                    domain: 'mock://www.child.com',
+                    props:  {
+                        account: {
+                            type:       'string',
+                            queryParam: false,
+                            required:   true,
+                            value:      ({ props }) => {
+                                return props.account;
```

Comment:
in here `props.account` was undefined. It's kind of a roundabout test, so I think we can clean it up to not test off this later going into a query param. I just wanted to make the test simulate the bug for the short term so if we do choose to merge this I'll update this to be more concise and clear.

Reply:
Having `props.account.value` return `props.account` is kinda weird. `value` is only intended for when the caller will not be passing in the value.

Reply:
Yeah I agree with it being confusing as well

Reply:
Ah, I see this is exactly what messaging-components is doing. Erk.

Reply:
yah! i think that might resolve the confusion on both our ends haha. I'm trying to make something goofy happening and you are seeing its goofy. Granted I think I described it poorly at first

Reply:
Yeah, ok, the more I look at it, this fix works for me.

Honestly, I kinda hate `extendProps`; it suffers from:

- Recycling the same mutable props object and extending it over and over
- It's never clear when you get `props` in one of these helper functions whether it's only partially constructed, or the default values are populated, or it's decorated, or not.

But I don't think there's much to be done about that right now. We should re-think/overhaul it for the next major version.

In the mean time `{ ...existingProps, ...inputProps }` sounds good.

Thanks for taking the time to investigate!

Reply:
Okay cool I think theres still a few flow errors and such to completely resolve and like I mentioned, I want to make the test read more clearly. I'll tackle those in the morning. 

Yeah super happy to chat more about an overhaul in the future. I'll need to dive deeper into zoid myself to provide more value on that conversation.

I'll mark the PR as "not draft" once its all cleaned up. If you think of anything in the meantime of course let me know and I'm happy to change direction! Thanks for chatting through this, and sorry for the super muddy description I initially provided!